File: src/Budgetify.Domain/BudgetPlans/BudgetPlan.cs
﻿using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CommonAbstractions.DB.Entities;
using DateKit.DB;
#pragma warning disable CS8618

namespace Budgetify.Domain.BudgetPlans
{
    public sealed class BudgetPlan : Entity<BudgetPlanId>
    {
        public DateTimeRange BudgetPeriod { get; private set; }
        public IReadOnlyCollection<BudgetedTransactionCategory> BudgetedTransactionCategories => _budgetedTransactionCategories;
        private readonly List<BudgetedTransactionCategory> _budgetedTransactionCategories;

        private BudgetPlan()
        { }

        public BudgetPlan(DateTimeRange budgetPeriod): base(new BudgetPlanId())
        {
            _budgetedTransactionCategories = new ();
            BudgetPeriod = budgetPeriod;
        }
    }
}

File: src/client-app/src/models/budgetPlans/budgetPlan.ts
import {BudgetedTransactionCategory} from "./budgetedTransactionCategory";
import {Transaction} from "../transactions/transaction";
import {DateTimeRange} from "../shared/dateTimeRange";

export interface BudgetPlan {
    budgetPlanId: string
    budgetPeriod: DateTimeRange
    budgetedTransactionCategories: BudgetedTransactionCategory[]
    transactions: Transaction[]
}